prog/kubernetes/kubernetes_cluster_nexus.rb

```diff
@@ -18,17 +18,38 @@ def self.assemble(name:, project_id:, location_id:, version: Option.kubernetes_v
 
       ubid = KubernetesCluster.generate_ubid
       subnet = if private_subnet_id
-        PrivateSubnet[id: private_subnet_id, project_id: Config.kubernetes_service_project_id] || fail("Given subnet is not available in the k8s project")
+        PrivateSubnet[id: private_subnet_id, project_id: project_id] || fail("Given subnet is not available in the project")
       else
+        # Will create customer private subnet with customer firewall
         Prog::Vnet::SubnetNexus.assemble(
-          Config.kubernetes_service_project_id,
+          project_id,
           name: "#{ubid}-subnet",
           location_id:,
           firewall_name: "#{ubid}-firewall",
           ipv4_range: Prog::Vnet::SubnetNexus.random_private_ipv4(Location[location_id], project, 18).to_s
         ).subject
       end
 
+      # Internal control plane node firewall, will be directly attached to kubernetes control plane VMs
+      internal_cp_vm_firewall = Firewall.create(name: "#{ubid}-cp-vm-firewall", location_id:, description: "Kubernetes control plane node internal firewall", project_id: Config.kubernetes_service_project_id)
+      internal_cp_vm_firewall.replace_firewall_rules([
+        {cidr: "0.0.0.0/0", port_range: Sequel.pg_range(22..22)},
+        {cidr: "::/0", port_range: Sequel.pg_range(22..22)},
+        {cidr: "0.0.0.0/0", port_range: Sequel.pg_range(443..443)},
+        {cidr: "::/0", port_range: Sequel.pg_range(443..443)},
+        {cidr: subnet.net4.to_s, port_range: Sequel.pg_range(10250..10250)},
+        {cidr: subnet.net6.to_s, port_range: Sequel.pg_range(10250..10250)}
+      ])
+
+      # Internal worker node firewall, will be directly attached to kubernetes worker VMs
+      internal_worker_vm_firewall = Firewall.create(name: "#{ubid}-worker-vm-firewall", location_id:, description: "Kubernetes worker node internal firewall", project_id: Config.kubernetes_service_project_id)
```

Comment:
cc @mohi-kalantari 

Just to understand the UX better, after this change, users would need to open specific ports (most commonly 80 and 443) of their worker vm firewall as an extra step while deploying apps, right? If so, we will need to update our docs and tutorials to reflect this change as well.

Reply:
Btw, AFAIK, we still allow all traffic within the subnet regardless the VM specific FW rules, right? If not, then we might break even the simples non-exposed app deployment flow.

Should we change an K8s E2E test to use the new flow with a customer subnet?